helix-term/src/compositor.rs

```diff
@@ -138,16 +139,27 @@ impl Compositor {
 
         // propagate events through the layers until we either find a layer that consumes it or we
         // run out of layers (event bubbling)
-        for layer in self.layers.iter_mut().rev() {
-            match layer.handle_event(event, cx) {
+        let event_results: Vec<_> = self
+            .layers
+            .iter_mut()
+            .rev()
+            .map(|layer| layer.handle_event(event, cx))
+            .collect();
```

Comment:
unfortunately, rust does non have an inclusive variant of `take_while` (itertools has tho)
does it matter that we call through all the layers even tho the event might already have been consumed or is it negligible?

Reply:
Yeah no, `handle_event` shouldn't be called on layers after a `Consumed` is returned or you will end up executing text editing commands while typing into a picker.

```rust
let callbacks = Vec::new();
for layer in self.layers.iter_mut().rev() {
   match layer.handle_event(event, cx) {
     Consumed(Some(callback)) => {
        callbacks.push(callback);
        break;
     }
     // ...
   }
}

// then execute all the callbacks in order
```